Source/BusinessMomentsAdaptorImage/Dockerfile

```diff
@@ -1,24 +1,44 @@
-# add stuff here
-FROM node:14.2
+FROM node:14.2.0 AS shared-build
 
-# Create app directory
 WORKDIR /app
 
-# Install app dependencies
-# A wildcard is used to ensure both package.json AND package-lock.json are copied
-# where available (npm@5+)
 COPY package.json ./
+COPY .eslintrc.js ./
+COPY tsconfig.json ./
+#COPY application.json ./
 
-RUN npm install -g typescript ts-node
-#RUN yarn add typescript
-#RUN yarn add ts-node
+COPY Source/Shared/Backend/package.json ./Source/Shared/Backend/package.json
+COPY Source/Shared/Common/package.json ./Source/Shared/Common/package.json
+COPY Source/Shared/Specs/package.json ./Source/Shared/Specs/package.json
+COPY Source/Shared/Styles/package.json ./Source/Shared/Styles/package.json
+COPY Source/Shared/Web/package.json ./Source/Shared/Web/package.json
 RUN yarn
 
-# If you are building your code for production
-# RUN npm ci --only=production
+#COPY Source/BusinessMomentsAdaptorImage/Backend/microservice.json /app/Source/BusinessMomentsAdaptorImage/microservice.json
 
-# Bundle app source
-COPY . .
+# Backend Build
+FROM node:14.2.0 AS backend-build
 
-EXPOSE 3008
-CMD [ "ts-node", "startserver.ts" ]
+WORKDIR /app
+
+COPY --from=shared-build /app/ ./
+
+WORKDIR /app/Source/BusinessMomentsAdaptorImage
+COPY Source/BusinessMomentsAdaptorImage/package.json ./package.json
+RUN yarn
+
+#COPY Source/Shared /app/Source/Shared/
+COPY Source/BusinessMomentsAdaptorImage/ ./
+
+RUN yarn build
+
+# Resulting Image
+FROM node:14.2.0-slim as base
+ARG CONFIGURATION=Release
+
+RUN echo Configuration = $CONFIGURATION
+
+WORKDIR /app
+COPY --from=backend-build /app/Source/BusinessMomentsAdaptorImage/dist/index.js ./
+
+ENTRYPOINT ["node", "index.js"]
```

Comment:
I checked this PR. It seems good. but can you explain how we can benefit from having js in the container? I supposed we would like to have TS as the base language in codes/containers. Maybe we can remove typescript installation from the dockerfile, with ts-node we still can be able to convert the TS into JS inside the container. Have you changed the dockerfile to use JS as the base language, in order to make the containers smaller/less complicated?

Reply:
Typescript is transpiled to JavaScript and then executed by nodejs. I'm not sure what you mean by suggesting to remove typescript install from the dockerfile. Can you elaborate? 

Yes, I have changed to use js files to make the container smaller.

Reply:
It's a good point to have smaller containers.
I supposed you were pointing to installing typescript inside the old dockerfile, when you said we don't need to install it. I meant we could remove it and leave the installation of ts-node as it works.
But in general, that makes sense to have the containers based on JS.